### PR TITLE
add events and analytics as soon as

### PR DESCRIPTION
**Pull request by jamesgorrie**
There is a few oddities in the way and order that we have decided to add events, thus analytics.
This fixes that.

At the moment we aren't actually tracking video plays if we preroll of on fronts. Useful.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
CC @Calanthe and @ScottPainterGNM

**Comment by OliverJAsh** [comment]
Lots good to me :+1:

**Comment by jamesgorrie** [comment]
@rich-nguyen - very good advice about moving autoplay, it also makes it a little more easy to fathom.

**Comment by jamesgorrie** [comment]
@OliverJAsh - it'd be good to get the once over again as I changed a bit.

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -204,49 +212,55 @@ define([
...
-                events.initOphanTracking(player, mediaId);
+                player.persistvolume({namespace: 'gu.vjs'});
 
                 // preroll for videos only
                 if (mediaType === 'video') {
-
                     player.fullscreener();
 
-                    if (withPreroll && !blockVideoAds) {
+                    if (showEndSlate && detect.isBreakpoint({ min: 'desktop' })) {
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: any reason not do this on portrait tablet?

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 2]
```diff
@@ -257,34 +271,35 @@ define([
...
-        mouseMoveIdle = debounce(function () { player.removeClass('vjs-mousemoved'); }, 500);
+                mouseMoveIdle = debounce(function () { player.removeClass('vjs-mousemoved'); }, 500);
 
-        // built in vjs-user-active is buggy so using custom implementation
-        player.on('mousemove', function () {
-            player.addClass('vjs-mousemoved');
-            mouseMoveIdle();
+                // built in vjs-user-active is buggy so using custom implementation
+                player.on('mousemove', function () {
+                    player.addClass('vjs-mousemoved');
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: fastdom?

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 3]
```diff
@@ -257,34 +271,35 @@ define([
...
                     });
-
-                    events.bindContentEvents(player);
+                    resolve();
                 }
-            });
-        });
 
-        mouseMoveIdle = debounce(function () { player.removeClass('vjs-mousemoved'); }, 500);
+                mouseMoveIdle = debounce(function () { player.removeClass('vjs-mousemoved'); }, 500);
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: fastdom

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 1 above)
Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Yes, we haven't tested it, and we're still uncertain about how ads optimise for slow connections.
We do need to revisit this, perhaps with commercial and another PR.

CC @kenlim

Comment (jamesgorrie) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: (As a side note, this was as it was before.)

Comment (sndrs) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: fair enough

**Comment by rich-nguyen** [comment]
cool :+1:

**Pull request merged**